test/e2e/providers/azure/azure.go

```diff
@@ -119,11 +124,218 @@ func (p *Provider) GetType() config.PlatformType {
+								{
+									Name:      "kubelet-dir",
+									MountPath: "/var/lib/kubelet",
+								},
+								{
+									Name:      "plugin-dir",
+									MountPath: "/csi",
+								},
+								{
+									Name:      "azure-config",
+									MountPath: "/k",
+								},
+								{
+									Name:      "csi-proxy-filesystem-v1",
+									MountPath: `\\.\pipe\csi-proxy-filesystem-v1`,
+								},
+								{
+									Name:      "csi-proxy-smb-pipe-v1",
+									MountPath: `\\.\pipe\csi-proxy-smb-v1`,
+								},
+							},
+						},
+					},
+					Volumes: []core.Volume{
+						{
+							Name: "csi-proxy-filesystem-v1",
+							VolumeSource: core.VolumeSource{
+								HostPath: &core.HostPathVolumeSource{
+									Path: `\\.\pipe\csi-proxy-filesystem-v1`,
+								},
+							},
+						},
+						{
+							Name: "csi-proxy-smb-pipe-v1",
+							VolumeSource: core.VolumeSource{
+								HostPath: &core.HostPathVolumeSource{
+									Path: `\\.\pipe\csi-proxy-smb-v1`,
+								},
+							},
+						},
+						{
+							Name: "registration-dir",
+							VolumeSource: core.VolumeSource{
+								HostPath: &core.HostPathVolumeSource{
+									Path: `C:\var\lib\kubelet\plugins_registry\`,
+									Type: &directoryType,
+								},
+							},
+						},
+						{
+							Name: "kubelet-dir",
+							VolumeSource: core.VolumeSource{
+								HostPath: &core.HostPathVolumeSource{
+									Path: `C:\var\lib\kubelet`,
+									Type: &directoryType,
+								},
+							},
+						},
+						{
+							Name: "plugin-dir",
+							VolumeSource: core.VolumeSource{
+								HostPath: &core.HostPathVolumeSource{
+									Path: `C:\var\lib\kubelet\plugins\file.csi.azure.com\`,
+									Type: &directoryOrCreate,
+								},
+							},
+						},
+						{
+							Name: "azure-config",
+							VolumeSource: core.VolumeSource{
+								HostPath: &core.HostPathVolumeSource{
+									Path: `C:\k\`,
+									Type: &directoryOrCreate,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	// See if the DaemonSet already exists in the state we expect it to be in.
+	existingDS, err := client.AppsV1().DaemonSets(csiNamespace).Get(context.TODO(), dsName, meta.GetOptions{})
+	if err != nil {
+		if !k8sapierrors.IsNotFound(err) {
+			return fmt.Errorf("error getting existing Windows CSI DaemonSet: %w", err)
+		}
+	} else if err == nil {
+		if reflect.DeepEqual(existingDS.Spec, ds.Spec) {
+			// DaemonSet is already as expected, nothing to do here.
+			return nil
+		}
+		// Delete the DaemonSet as it has the wrong spec.
+		err = client.AppsV1().DaemonSets(csiNamespace).Delete(context.TODO(), dsName, meta.DeleteOptions{})
+		if err != nil {
+			return fmt.Errorf("error deleting existing Windows CSI DaemonSet: %w", err)
+
+		}
+	}
+	_, err = client.AppsV1().DaemonSets(csiNamespace).Create(context.TODO(), &ds, meta.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("error creating Windows CSI DaemonSet: %w", err)
+	}
+	return nil
```

Comment:
Do you think we should check the containers are running before returning?

Reply:
I don't think that's necessary, if the containers don't go to running the workload won't be able to deploy so the failure will be shown there.